notifications.yaml

```diff
@@ -0,0 +1,220 @@
+        type: "string"
+        format: "uuid"
+    get: 
+      operationId: "listMessagesByUser"
+      summary: "List Messages by user"
+      tags: 
+        - "Messages"
+      description: "Retrive all messages for this user. \n\nMessages will be returned in inverse delivery order (from last to first)"
+      responses: 
+        200: 
+          description: "Found."
+          schema: 
+            $ref: "#/definitions/MessageCollection"
+  /Messages: 
+    post: 
+      operationId: "sendMessage"
+      summary: "Send Message"
+      tags: 
+        - "Messages"
+      description: "Submit a message for delivery."
+      parameters: 
+        - 
+          name: "message"
+          in: "body"
+          schema: 
+            $ref: "#/definitions/MessageDefinition"
+      responses: 
+        202: 
+          description: "Message accepted for delivery."
+          schema: 
+            allOf: 
+              - 
+                properties: 
+                  id: 
+                    type: "string"
+                    format: "uuid"
+                  status: 
+                    $ref: "#/definitions/MessageStatus"
+                required: 
+                  - "id"
+                  - "status"
+            description: ""
+            type: "object"
+          headers: 
+            Location: 
+              type: "string"
+              description: "Location (URL) of accepted message resource.\nSend a GET request to this URL to retrieve the message status and details."
+definitions: 
+  ProblemJson: 
+    title: "Problem Type"
+    type: "object"
+    properties: 
+      type: 
+        type: "string"
+        format: "uri"
+        description: "An absolute URI that identifies the problem type.  When dereferenced,\nit SHOULD provide human-readable documentation for the problem type\n(e.g., using HTML).\n"
+        default: "about:blank"
+        example: "https://example.com/problem/constraint-violation"
+      title: 
+        type: "string"
+        description: "A short, summary of the problem type. Written in english and readable\nfor engineers (usually not suited for non technical stakeholders and\nnot localized); example: Service Unavailable\n"
+      status: 
+        type: "integer"
+        format: "int32"
+        description: "The HTTP status code generated by the origin server for this occurrence\nof the problem.\n"
+        minimum: 100
+        maximum: 600
+        exclusiveMaximum: true
+        example: 503
+      detail: 
+        type: "string"
+        description: "A human readable explanation specific to this occurrence of the\nproblem.\n"
+        example: "Connection to database timed out"
+      instance: 
+        type: "string"
+        format: "uri"
+        description: "An absolute URI that identifies the specific occurrence of the problem.\nIt may or may not yield further information if dereferenced."
+  MessageContent: 
+    title: "MessageContent"
+    type: "object"
+    properties: 
```

Comment:
this is a discussion we already had - we have to chose one of these two options:

1. having the sender explicitly decide the destination channel and providing a message formatted for that specific channel (i.e. subject and body for email, short text for sms, etc...)
2. hiding the channel decision in the system, making it based on user preferences, hiding it from the sender, having some logic that takes the message and formats it for the channel

since `postMessage` doesn't allow setting a channel, I'm assuming we're going for option 2

in this second option, we decided to hide the channels from the sender, so we can either:

1. set for the amount of information that satisfies the "simplest" channel (e.g. SMS) and upgrade this content for richer channels (e.g. email) - this is fairly simple to do (e.g. display a simple pre-formatted text in an email) 
2. set for the the amount of information that satisfies the "richest" channel (e.g. HTML email) and downgrade this content for simpler channels (e.g SMS) - this is harder to do (how to you fit an HTML email in an SMS?) - an imperfect option would be to add indirection (e.g. send a link by SMS pointing to the HTML email)
3. option in between, ask for different formats of the same message without leaking the underlying abstractions (i.e. we may add/remove channels in the future, so we don't want to talk about specific channels like email, sms or push notifications, instead we want to talk about different formats for the same content, think about a kind of responsive version of the message, e.g. short text or rich long text, etc...)